### ISSUE TITLE
Multiple in/out of geofence messages when using TK103A tracker

### ISSUE THREAD

**Issue by geralddavison** [vitalidze/traccar-web#1090]
The TK103A vehicle tracker stores positions taken when out of mobile phone network coverage.

When it next enters mobile phone network coverage it starts to send the stored location points.

When the next schedule point is taken it sends this point immediately.

If it still has stored points to send then these will then follow.  This will cause the location to "jump back" to a previous position.

If the stored positions are within a geofence, but the current positions are not, it has the effect of rapidly moving in and out of the geofence causing multiple exit/enter messages.  The times stamps within the points make it clear when the point was taken.  But it appears traccar is taking the time of trackpoint delivery to work out entry/exit events.

**Comment by vitalidze** [comment]
> But it appears traccar is taking the time of trackpoint delivery to work out entry/exit events. 

That's incorrect. It takes time sent by device. It is never using other times for any report in the system.

However, there is one notable thing regarding posting events for notifications. For posting events it scans all received positions sorted by identifiers. This means they are always scanned in order they were received. So, if for some reason your device sends data in non-chronological order, then you may start receiving unexpected notifications.
